os/arch/arm/src/common/up_createstack.c

```diff
@@ -198,7 +198,12 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		 * in user threads and null for kernel threads.
 		 */
 		if (!uheap) {
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+			stack_alloc_size = stack_alloc_size + CONFIG_MPU_STACK_GUARD_SIZE;
+			tcb->stack_alloc_ptr = (uint32_t *)kmm_memalign(CONFIG_MPU_STACK_GUARD_SIZE, stack_alloc_size);
```

Comment:
For ARMv8m, we don't need to make alignment for mpu region. How about adding ifdef for checking v8m?

Reply:
@jeongchanKim , All MPU regions are aligned to a multiple of 32 bytes in ARMV8m, so the above condition  should be fine

Reply:
Oh okay. I mistook. Thanks.

Reply:
About this, you provide the config, CONFIG_MPU_STACK_GUARD_SIZE and it could have another value. When user sets it as another value, not 32, it wastes memory. When V8M is enabled, could you make the alignment is fixed as 32?
```
FAR void *memalign(size_t alignment, size_t size)
```

Reply:
What I am saying is that even stack guard is bigger than 32, allocation allignment for v8m can be 32 fixed value always.
But currently it does not work like that. When user changes the guard size, alignment value is also changed.

Reply:
Currently, allocation alignment in `up_create_stack `for user thread is **not conditional to v8m**.
```
/* Use the user-space allocator if this is a task or pthread */
#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION	/* address must be aligned to 32 byte for MPU region */
			stack_alloc_size = stack_alloc_size + CONFIG_MPU_STACK_GUARD_SIZE;
			tcb->stack_alloc_ptr = (uint32_t *)kumm_memalign(CONFIG_MPU_STACK_GUARD_SIZE, stack_alloc_size);
#else
			tcb->stack_alloc_ptr = (uint32_t *)kumm_malloc(stack_size);
#endif
```

I will make allocation alignment (for both kernel/user thread) as fixed value 32 bytes always as per your suggestion.
But can you confirm if this change is to be specific only for v8m?

Reply:
Made v8m specific changes here as recommended.